### ISSUE TITLE
False positive `IDE0241 Nullable directive is unnecessary`

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#75951]
**Version Used**: 
```
Microsoft Visual Studio Professional 2022
Version 17.13.0 Preview 1.0
VisualStudio.17.Preview/17.13.0-pre.1.0+35507.96
Microsoft .NET Framework
Version 4.8.09037
```

**Steps to Reproduce**:

Paste the following code into the editor
```
#nullable enable

public class Test
{
#nullable disable
  public Test() { }
#nullable enable

  public string Value { get; set; }
}
```

**Diagnostic Id**:

`IDE0241 Nullable directive is unnecessary` is reported for `#nullable disable` directive

**Expected Behavior**:
The directive is not reported as redundant as it turns off nullable analysis warnings in the constructor

**Actual Behavior**:
`IDE0241 Nullable directive is unnecessary` is reported for `#nullable disable` directive
The directive is highlighted as redundant
Removing the directive introduces a compiler warning `CS8618: Non-nullable property `Value` must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.`

**Comment by arunchndr** [comment]
There was a fix that went out for this in 17.12.3 tracked via https://developercommunity.visualstudio.com/t/Private-Unity-messages-incorrectly-marke/10779025. Closing as duplicate.
